### PR TITLE
@orta => Reboot fastlane integration

### PR DESCRIPTION
**Pull request by alloy**

### CODE REVIEW

**Comment by orta** [comment]
Soooooooooo gooooood

**Comment by alloy** [comment]
I had wanted to stitch the iPhone 4" images together vertically and use it in the README, but having to install ImageMagick just for this was too much for me.

In the future we could maybe create a Swift one-liner that does it, but for now I’m calling this done.

File: Gemfile.lock [hunk 1]
```diff
@@ -203,50 +132,29 @@ GEM
       cliver (~> 0.3.1)
       multi_json (~> 1.0)
       websocket-driver (>= 0.2.0)
-    produce (0.2.1)
-      fastlane_core (>= 0.5.0)
+    pry (0.10.2)
```
Comment (orta) [inline Gemfile.lock]: woo

File: fastlane/DeliverFile [hunk 2]
```diff
@@ -0,0 +1,67 @@
+app_identifier 'net.artsy.artsy'
+username 'it@artsymail.com'
+
+# PATCH
+::FastlaneCore::Languages::ALL_LANGUAGES << 'zh-Hans'
+
+module ::Deliver
```
Comment (orta) [inline fastlane/DeliverFile]: this feels like core functionality, can you add some comments as to _why_ its in here?

**Comment by orta** [comment]
one blocking comment, otherwise looks good.

@KrauseFx will want to see this :+1:

**Comment by KrauseFx** [comment]
Thanks for the mention @orta. @alloy what's up with the monkey patching of `deliver` upload feature? Is something broken? I'm more than happy to build that in into the `deliver` gem if that's applicable for all users.

File: Gemfile.lock (hunk 1 above)
Comment (alloy) [inline Gemfile.lock]: I don’t know why pry is needed.

**Comment by alloy** [comment]
@KrauseFx Currently `deliver` assumes that all directories in the `metadata` and `screenshots` dirs are language dirs, but in our case we have `common`, so that doesn’t apply. These monkey-patches only operate on supported languages.

File: Gemfile.lock (hunk 1 above)
Comment (KrauseFx) [inline Gemfile.lock]: Interesting, this shouldn't be caused by `fastlane`... the `fastlane` repos do have `pry` as development dependency though to make it easy for contributors to debug issues.

**Comment by orta** [comment]
> Currently deliver assumes that all directories in the metadata and screenshots dirs are language dirs, but in our case we have common, so that doesn’t apply. These monkey-patches only operate on supported languages.

Add comments to this in the code, and t's shippable :+1:

**Comment by alloy** [comment]
I’ll first make PRs to `deliver`/`fastlane`, once I’ve had the time to do that I’ll update this PR for merging.

**Comment by orta** [comment]
ok, cool

**Comment by orta** [comment]
this could do with a `make store` or something for the specific `deliver` command

**Comment by alloy** [comment]
> this could do with a make store or something for the specific deliver command

I’m unsure, the problem is that we don’t always want it to upload screenshots etc. I’m open to suggestions on how to otherwise handle this.

**Comment by alloy** [comment]
Merging this before it gets too old.

**Pull request merged**